tests/test_crawlers.py

```diff
@@ -390,6 +415,30 @@ def test_intersects_time_range_no_upper_limit(self):
         # no upper limit and no start_time, without intersection
         self.assertFalse(crawler._intersects_time_range(None, datetime(2019, 2, 19)))
 
+    @mock.patch("geospaas_harvesting.crawlers.OpenDAPCrawler._http_get")
+    @mock.patch("geospaas_harvesting.crawlers.OpenDAPCrawler._get_links")
+    def test_correct_navigation_to_download_page_for_OSISAF(self, mock_get_link, mock_http_get):
+        """Test the functionality of "get_download_url" method for OpenDAP crawler of OSISAF project """
+        mock_get_link.return_value = [
+            '/thredds/dodsC/osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc.html']
+        expected_urls = ['https://thredds.met.no/thredds/catalog/osisaf/met.no/ice/amsr2_conc/2019/11/catalog.html?dataset=osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc', ]
+        request_link = crawlers.ThreddsCrawler.get_download_url(
+            crawlers.OpenDAPCrawler, expected_urls)
```

Comment:
Why not just instantiate a `ThreddsCrawler` object instead of accessing `get_download_url()` in this very confusing way?

Reply:
@aperrin66 
This test must  show how this function works! Crystal clear of what comes in! and what goes out of this function! Those ` return values` and `expected_urls` are real case values in a harvesting process. So it worth it to write in this way I think.

Reply:
I wasn't talking about that.
You are accessing an instance method directly from the class.

The call to `get_download_url()` should be written this way:
`crawlers.ThreddsCrawler().get_download_url(expected_urls)`

Reply:
yeah you are right!

Reply:
@aperrin66 
actually that impletementation give us this error:
![image](https://user-images.githubusercontent.com/62345618/90616666-ffa9ed00-e20d-11ea-9375-8c66c80f8da0.png)


Which indicates the demand of proper instantiation!
and if we put this instead:
```python
request_link = crawlers.ThreddsCrawler(root_url='/thredds/dodsC/osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc.html').get_download_url(expected_urls)

```
or
```python
request_link = crawlers.ThreddsCrawler(root_url='').get_download_url(expected_urls)
```

It will give us this error:
![image](https://user-images.githubusercontent.com/62345618/90616933-5d3e3980-e20e-11ea-948d-b97035e6dcd5.png)

Reply:
SO I think it is better to avoid engaging with instantiation and only focus on this function in this test.right?

Reply:
You get this error because you mock OpenDAPCrawler but you use ThreddsCrawler. Here is a version that works:
```python
    @mock.patch("geospaas_harvesting.crawlers.ThreddsCrawler._http_get")
    @mock.patch("geospaas_harvesting.crawlers.ThreddsCrawler._get_links")
    def test_correct_navigation_to_download_page_for_OSISAF(self, mock_get_link, mock_http_get):
        """Test the functionality of "get_download_url" method for OpenDAP crawler of OSISAF project """
        mock_get_link.return_value = [
            '/thredds/dodsC/osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc.html']
        catalog_url = 'https://thredds.met.no/thredds/catalog/osisaf/met.no/ice/amsr2_conc/2019/11/catalog.html?dataset=osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc'
        request_link = crawlers.ThreddsCrawler('').get_download_url(catalog_url)
        self.assertEqual(
            request_link, 'https://thredds.met.no/thredds/dodsC/osisaf/met.no/ice/amsr2_conc/2019/11/ice_conc_nh_polstere-100_amsr2_201911301200.nc.dods')
```

Reply:
Done in the next commit

Reply:
Ooooh!!! my bad.

Done in the next commit